native/cocos/platform/android/AndroidPlatform.cpp

```diff
@@ -567,6 +567,7 @@ int AndroidPlatform::init() {
 void AndroidPlatform::onDestroy() {
     UniversalPlatform::onDestroy();
     unregisterAllInterfaces();
+    JniHelper::onDestroy();
```

Comment:
It's too early to destroy jni env;  jni method after may be called another thread finish its task.

Reply:
Consider calling jni's destroy on the finalize interface of activity

Reply:
In JniHeper::onDestroy not destroyed jni env, just release activity's global reference

Reply:
> activity's

classloader may be used after calling destroy

Reply:
> activity's

if activity has destroyed, it should not has any jni call from engine api, because at this time engine has paused->destroyed. Developer has no chance call any engine api associated with jni.

Reply:
if engine internal has chance to use classloader  after destroyed, i think we should avoid or solve this condition.

Reply:
> if engine internal has chance to use classloader after destroyed, i think we should avoid or solve this condition.

after Activity's destroy called, download tasks may be need to call cancel, thus classloader need to be used

Reply:
We can release jniObjectActivity only here, classloader's lifecycle is the same as application

Reply:
> We can release jniObjectActivity only here, classloader's lifecycle is the same as application

if download tasks cannot cancel before destory, JniHelper::classloader must add double check to avoid NewGlobalRef many times.

Reply:
> We can release jniObjectActivity only here, classloader's lifecycle is the same as application

new commit
improve:only release jniObjectActivity,keep classloader
add nullptr check for classloader to avoid new global reference many times